common/java_deps/rules.bzl

```diff
@@ -108,28 +108,30 @@ _transitive_collect_maven_coordinate = aspect(
 
 def _java_deps_impl(ctx):
     names = {}
-    files = []
-    filenames = []
-    outputPathOverrides = ctx.attr.java_deps_root_overrides
+    files_by_output_path = {}
+    output_path_overrides = ctx.attr.java_deps_root_overrides
 
     mapping = ctx.attr.target[TransitiveJarToMavenCoordinatesMapping].mapping
 
-    for file in ctx.attr.target.data_runfiles.files.to_list() + ctx.attr.target.files.to_list():
```

Comment:
It looks like we added `ctx.attr.target.files` here a couple of years ago, in order to support a particular scenario that we ended up not using - so now we're cleaning it up.